content/dota_addons/legion_td/panorama/scripts/custom_game/team_select_player.js

```diff
@@ -30,10 +30,17 @@ function OnPlayerDetailsChanged() {
 
 function SetTitle() {
     var title = "";
-    title = $.Localize("#legion_contributor_leader");
+    if (Contains(GameUI.CustomUIConfig().PreviousLeaders, steam_id)) {
+        title = $.Localize("#legion_contributor_previous_leader");
+    }
     $("#PlayerTitle").text = title;
 }
 
+function Contains(array, toSearch) {
+    var result;
+    array.forEach(function (entry) { if (entry == toSearch) result = true; });
+    return result;
+}
```

Comment:
This logic should be somewhere else (maybe a seperate file that is imported in `team_select` as well as `main_hud`)